### PR TITLE
replace the local setup tutorial with a landscaper-cli quickstart tutorial

### PR DESCRIPTION
**Pull request by reshnm**
**How to categorize this PR?**

/area documentation
/kind enhancement
/priority 4

**What this PR does / why we need it**:

The local installtion of the landscaper and the harbour OCI registry as descibed in the local setup tutorial is very error prone.
This PR replaces the local setup tutorial with a tutorial that uses the `landscaper-cli quickstart` command, which is much easier to use.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: docs/tutorials/00-landscaper-installation.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
+# Install the Landscaper
+
+This tutorial describes how to install the Landscaper on a Kubernetes cluster and optionally how to install an OCI registry
+on a Gardener Shoot cluster.
+
+### Prerequisites
+
+- Kubernetes cluster
+- Landscaper CLI (see the [installation documentation](https://github.com/gardener/landscapercli/blob/master/docs/installation.md))
+- This tutorial assumes that the landscaper will be installed in the namespace `landscaper`. The namespace has to created before the landscaper is being installed.
```
Comment (schrodit) [inline docs/tutorials/00-landscaper-installation.md:10]: in our examples and tests we use the namespace `ls-system`. Where is this namespace defined? in the cli?
We should stick to one to not confuse people.

**Review by schrodit — CHANGES_REQUESTED** [review]

**Review by schrodit — APPROVED** [review]

**Comment by reshnm** [comment]
I have added the deployer management configuration.
If the namespace is `ls-system` it will also work without the default configuration.
However, when a different namespace is specified, an explicit configuration is mandatory.

File: docs/tutorials/00-landscaper-installation.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
+# Install the Landscaper
+
+This tutorial describes how to install the Landscaper on a Kubernetes cluster and optionally how to install an OCI registry
+on a Gardener Shoot cluster.
+
+### Prerequisites
+
+- Kubernetes cluster
+- Landscaper CLI (see the [installation documentation](https://github.com/gardener/landscapercli/blob/master/docs/installation.md))
+- This tutorial assumes that the landscaper will be installed in the namespace `ls-system`. The namespace has to created before the landscaper is being installed.
```
Comment (achimweigel) [inline docs/tutorials/00-landscaper-installation.md:10]:
```suggestion
- This tutorial assumes that the landscaper will be installed in the namespace `ls-system`. The namespace has to be created before the landscaper is being installed.
```

File: docs/tutorials/00-landscaper-installation.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+This tutorial uses the `landscaper-cli` `quickstart` command. For additional details please see the [documentation](https://github.com/gardener/landscapercli/blob/master/docs/commands/quickstart/install.md).
+
+#### Alternative 1: Install with OCI registry
+
+The `landscaper-cli` can install the Landscaper together with an OCI registry. The OCI registry will be exposed by an ingress.
+
+##### Prerequisites
+
+- The target cluster must be a Gardener Shoot (TLS is provided via the Gardener cert manager).
```
Comment (achimweigel) [inline docs/tutorials/00-landscaper-installation.md:22]:
```suggestion
- The target cluster must be a Gardener Shoot (TLS certificate is provided via the Gardener cert manager). 
```

File: docs/tutorials/00-landscaper-installation.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+      deployerManagement:
+        disable: false
+        namespace: ls-system
+        agent:
+          disable: false
+          namespace: ls-system
+```
+
+This tutorial assumes that the OCI registry usernames is store in environment variable `REGISTRY_USER` and the password is stored in environment variable `REGISTRY_PASSWORD`.
```
Comment (achimweigel) [inline docs/tutorials/00-landscaper-installation.md:49]:
```suggestion
This tutorial assumes that the OCI registry username is stored in environment variable `REGISTRY_USER` and the password is stored in environment variable `REGISTRY_PASSWORD`.
```

File: docs/tutorials/00-landscaper-installation.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+      deployerManagement:
+        disable: false
+        namespace: ls-system
+        agent:
+          disable: false
+          namespace: ls-system
+```
+
+The registry secrets have to be provided in a plain Docker auth JSON format. See [here](../gettingstarted/install-landscaper-controller.md#Private registry access) for more details.
```
Comment (achimweigel) [inline docs/tutorials/00-landscaper-installation.md:123]: The referenced documentation does not describe which section in `landscaper-values.yaml` to use for configuring the registry secrets. Here you find `landscaper.landscaper.registrySecrets` but this differs from `landscaper.landscaper.registryConfig.secrets`.

File: docs/tutorials/00-landscaper-installation.md [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+
+To install the Landscaper and the OCI container registry, use the following command:
```
Comment (achimweigel) [inline docs/tutorials/00-landscaper-installation.md:125]:
```suggestion
To install the Landscaper with the credentials for the OCI container registry, use the following command:
```

**Review by achimweigel — CHANGES_REQUESTED** [review]
This is a very good improvement for our documentation. We need much more of this. 

I only not sure if it is a good idea to have this tutorial in the landscaper repo and a similar one in the landscaper-cli repo?

File: docs/tutorials/00-landscaper-installation.md (hunk 5 above)
Comment (reshnm) [reply docs/tutorials/00-landscaper-installation.md:123]: It seems that the refernced documentation is wrong with `landscaper.landscaper.registrySecrets`.
`landscaper.landscaper.registryConfig.secrets`is correct.
I will fix the referenced doc in a separate PR.

**Comment by reshnm** [comment]
> This is a very good improvement for our documentation. We need much more of this.
> 
> I only not sure if it is a good idea to have this tutorial in the landscaper repo and a similar one in the landscaper-cli repo?

I agree, that there are some overlappings with the landscaper-cli docs. Currently I think we have everythink documented but it is scattered accross different repos and docs. I think it is a good thing to have one doc that describes how to install the landscaper so that it can be used for the following tutorials.

File: docs/tutorials/00-landscaper-installation.md [hunk 7]
```diff
@@ -0,0 +1,138 @@
+# Install the Landscaper
+
+This tutorial describes how to install the Landscaper on a Kubernetes cluster and optionally how to install an OCI registry
+on a Gardener Shoot cluster.
+
+### Prerequisites
+
+- Kubernetes cluster
```
Comment (schrodit) [inline docs/tutorials/00-landscaper-installation.md:8]: isnt' it also required for the cluster to be a shoot when the registry is installed?
Or alternativly a ingress and a certificate management:

But I think we should mention that

**Review by schrodit — CHANGES_REQUESTED** [review]

File: docs/tutorials/00-landscaper-installation.md (hunk 7 above)
Comment (reshnm) [reply docs/tutorials/00-landscaper-installation.md]: Yes, might be a bit confusing.
I have moved this prerequisite to alternative 2.
Alternative 1 requires a gardener shoot cluster with nginx ingress installed.
Alternative 2 expects an external OCI registry, therefore no ingress or cert manager is required.

**Review by schrodit — APPROVED** [review]

**Review by achimweigel — APPROVED** [review]
/lgtm

**Pull request merged**